src/snowcli/cli/stage/manager.py

```diff
@@ -33,17 +33,48 @@ def put(
         parallel: int = 4,
         overwrite: bool = False,
     ) -> SnowflakeCursor:
+        *_, put_cursor = self._put(
+            local_path=local_path,
+            stage_path=stage_path,
+            parallel=parallel,
+            overwrite=overwrite,
+        )
+        return put_cursor
+
+    def _put(
```

Comment:
I'm not sure if it is ok to add `_put` and `_remove` to StageManager. I think it will be better to move it to NativeAppManager

Reply:
Since `put` essentially calls `_put` but without any role, if I separate `put` and `_put` into different classes, it would essentially mean duplicating code across the two classes (with small differences). Would that be okay?

Reply:
An update to this:
1. `put` and `_put`, and `remove` and `_remove` have been separated out into their own commands, instead of adding an overload.
2. But I would like to still keep them in this file as they are doing what `put` and `remove` are, just with additional role changing. Additionally, other contributors might discover these two and find them fitting for their own use case as well.